kratos/geometries/geometry_dimension.h

```diff
@@ -0,0 +1,262 @@
+        : mDimension( rOther.mDimension )
+        , mWorkingSpaceDimension( rOther.mWorkingSpaceDimension )
+        , mLocalSpaceDimension( rOther.mLocalSpaceDimension )
+    {
+    }
+
+    /// Destructor. Do nothing!!!
+    virtual ~GeometryDimension() {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    GeometryDimension& operator=( const GeometryDimension& rOther )
+    {
+        mDimension = rOther.mDimension;
+        mWorkingSpaceDimension = rOther.mWorkingSpaceDimension;
+        mLocalSpaceDimension = rOther.mLocalSpaceDimension;
+
+        return *this;
+    }
+
+    ///@}
+    ///@name Informations
+    ///@{
+
+    /** Dimension of the geometry for example a triangle2d is a 2
+    dimensional shape. A line is 1 dimensional shape and
+    a point 0 dimensional.
+
+    @return SizeType, dimension of this geometry.
+    @see WorkingSpaceDimension()
+    @see LocalSpaceDimension()
+    */
+    inline SizeType Dimension() const
+    {
+        return mDimension;
+    }
+
+    /** Working space dimension. for example a triangle as part
+    of a surface is a 2 dimensional shape but can be used in
+    3 dimensional space.
+
+    @return SizeType, working space dimension of this geometry.
+    @see Dimension()
+    @see LocalSpaceDimension()
+    */
+    inline SizeType WorkingSpaceDimension() const
+    {
+        return mWorkingSpaceDimension;
+    }
+
+    /** Local space dimension. for example a triangle is a 2
+    dimensional shape but can have 3 dimensional area
+    coordinates l1, l2, l3.
+
+    @return SizeType, local space dimension of this geometry.
+    @see Dimension()
+    @see WorkingSpaceDimension()
+    */
+    inline  SizeType LocalSpaceDimension() const
+    {
+        return mLocalSpaceDimension;
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const
+    {
+        return "geometry dimension";
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo( std::ostream& rOStream ) const
+    {
+        rOStream << "geometry dimension";
+    }
+
+    /// Print object's data.
+    virtual void PrintData( std::ostream& rOStream ) const
+    {
+        rOStream << "    Dimension               : " << mDimension << std::endl;
+        rOStream << "    Working space dimension : " << mWorkingSpaceDimension << std::endl;
+        rOStream << "    Local space dimension   : " << mLocalSpaceDimension;
+    }
+
+    ///@}
+
+protected:
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    SizeType mDimension;
+
+    SizeType mWorkingSpaceDimension;
+
+    SizeType mLocalSpaceDimension;
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    virtual void save( Serializer& rSerializer ) const
+    {
+        rSerializer.save( "Dimension", mDimension );
+        rSerializer.save( "Working Space Dimension", mWorkingSpaceDimension );
```

Comment:
I'm not sure if save/load tags can have spaces. Is there any test for that?

Reply:
I change it.